src/utils/memo.ts

```diff
@@ -0,0 +1,19 @@
+/**
+ * For running something once
+ * @example
+ * const getProvider = once((ctx) => new R2Provider({ ctx }));
+ * // later...
+ * getProvider(ctx).something();
+ */
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+export function once<T, Args extends Array<any>>(
+  fn: (...args: Args) => T
+): (...args: Args) => T {
+  let value: T;
+  return (...args: Args) => {
+    if (value === undefined) {
```

Comment:
What if the real function returns `undefined` (?)

Reply:
I believe instead of checking value = undefined, create a 2nd variable named `hasBeenCalled` and set once you call and if it is true, then don't call the function again :)